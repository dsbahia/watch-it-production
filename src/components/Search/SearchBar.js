import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { toast } from "react-hot-toast";
import api from "../../api/api";
import "../../styles/searchbar.css";

function SearchBar({ setSearchResults }) {
  const [value, setValue] = useState("");
  const navigate = useNavigate();

  const handleSubmit = async (event) => {
    event.preventDefault();

    if (value.trim() === "") {
      const errorMsg = "Please enter a valid search term.";
      toast.error(errorMsg, {
        duration: 5000,
      });
      return;
    }

    const data = await api.searchMovies(value);
    if (data === null) {
      const errorMsg = "An error occurred. Please try again later.";
      toast.error(errorMsg, {
        duration: 5000,
      });
    } else {
      setSearchResults(data);

      navigate("/search");
    }
  };

  return (
    <div className="search-bar">
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          placeholder="Search"
          onChange={(e) => setValue(e.target.value)}
          data-testid="search-input"
        />
        <button className="search-button" type="submit">
          Go!
        </button>
      </form>
    </div>
  );
}

export default SearchBar;
