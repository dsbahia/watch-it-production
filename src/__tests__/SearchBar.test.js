import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SearchBar from "../components/Search/SearchBar";

describe("SearchBar", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SearchBar />
    </MemoryRouter>,
  );

  it("renders  correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("updates the value when user types", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <SearchBar />
      </MemoryRouter>,
    );
    const inputElement = getByTestId("search-input");

    fireEvent.change(inputElement, { target: { value: "test input" } });
    expect(inputElement.value).toBe("test input");
  });
});
