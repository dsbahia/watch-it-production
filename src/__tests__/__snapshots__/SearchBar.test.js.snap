// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders  correctly 1`] = `
<DocumentFragment>
  <div
    class="search-bar"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        data-testid="search-input"
        placeholder="Search"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
